src/core/kmod-setup.c

```diff
@@ -168,6 +168,15 @@ int kmod_setup(void) {
                 /* Make sure we can send sd-notify messages over vsock as early as possible. */
                 { "vmw_vsock_virtio_transport", NULL,                        false, false, has_virtio_vsock   },
 
+                /* We can't wait for specific virtiofs tags to show up as device nodes so we have to load the
+                 * virtiofs and virtio_pci modules early to make sure the virtiofs tags are found when
+                 * sysroot.mount is started.
+                 *
+                 * TODO: Remove these again once https://gitlab.com/virtio-fs/virtiofsd/-/issues/128 is
+                 * resolved and the kernel fix is widely available. */
+                { "virtiofs",                   "/sys/module/virtiofs",      false, false, NULL               },
```

Comment:
shouldn't this do something like the has_virtio_xyz() functions?

Reply:
Right but what would we check for? I guess we could check for "rootfstype=virtiofs" but that seems wrong since there might be other early mounts using virtiofs instead of just the root mount.

I grepped /sys and couldn't find anything in there either that we could check for

Reply:
Ah nvm these do have the usual modalias will fix